pkg/services/business_case.go

```diff
@@ -168,6 +170,43 @@ func NewUpdateBusinessCase(
 				Operation: apperrors.QuerySave,
 			}
 		}
+		// TODO: what is the best way to detect the step that this request is in? Current step? Or should we depend on what the current step is?
+		intake, err := fetchIntake(ctx, businessCase.SystemIntakeID)
+		if err != nil {
+			logger.Error("failed to update business case state")
+			return businessCase, &apperrors.QueryError{ //return the error
+				Err:       err,
+				Model:     businessCase,
+				Operation: apperrors.QuerySave,
+			}
+		}
+		/*
+			if newIntakeStatus == models.SystemIntakeStatusBIZCASEFINALSUBMITTED {
+				intake.FinalBusinessCaseState = models.SIRFSSubmitted
+			} else if newIntakeStatus == models.SystemIntakeStatusBIZCASEDRAFTSUBMITTED {
+				intake.DraftBusinessCaseState = models.SIRFSSubmitted
+			}
+		*/
+		// TODO: is there a better way to detect if this is a draft business case or a final business case?
+		if intake.Step == models.SystemIntakeStepDRAFTBIZCASE {
```

Comment:
Since the db doesn't differentiate between draft or final, we need to rely on the step the intake is in. If the intake isn't in that state, the business case state won't update.

Reply:
Can this conversation be resolved? Looks like the TODO comment's been removed. It might be helpful to put your reasoning above into a comment on the code.